packages/aws-cdk-lib/aws-ec2/lib/instance.ts

```diff
@@ -514,6 +527,20 @@ export class Instance extends Resource implements IInstance {
       throw new Error('You can\'t set both `enclaveEnabled` and `hibernationEnabled` to true on the same instance');
     }
 
+    if (
+      props.ipv6AddressCount !== undefined &&
+      !Token.isUnresolved(props.ipv6AddressCount) &&
+      (props.ipv6AddressCount < 0 || !Number.isInteger(props.ipv6AddressCount))
+    ) {
+      throw new Error(`\'ipv6AddressCount\' must be a non-negative integer, got: ${props.ipv6AddressCount}`);
+    }
+
```

Comment:
Curious that if there's a upper bound value for this property?

Reply:
I tried to create an instance by specifying `ipv6AddressCount` as 2000.

Result:
```sh
Failed resources:
InstanceIpv6AddressCountTestStack | 8:04:58 AM | CREATE_FAILED        | AWS::EC2::Instance                    | Instance (InstanceC1063A87) Resource handler returned message: "Address count 2000 per interface exceeds the limit for m5.large (Service: Ec2, Status Code: 400, Request ID: 6fcb0c3e-a029-4b62-9e31-4452642bede8)" (RequestToken: 498ff895-1532-848d-2a6b-2e2af377efb0, HandlerErrorCode: InvalidRequest)
```

It seems that the upper limit varies depending on the instance type. However, I couldn't find any documentation on this...

Reply:
Thanks for giving it a try. To be clear I don't want us to do the validation, but just curious about this.